pkg/k8s/watchers/cilium_envoy_config.go

```diff
@@ -85,14 +86,24 @@ func (k *K8sWatcher) ciliumEnvoyConfigInit(ctx context.Context, ciliumNPClient c
 	k.k8sAPIGroups.AddAPI(k8sAPIGroupCiliumEnvoyConfigV2)
 }
 
-// isCiliumIngress returns true if the given object metadata indicates that the owner needs the Envoy listener to assume the identity of Cilium Ingress. Currently this is the case when any of the OwnerReferences is Kind "Ingress" or "Gateway".
-func isCiliumIngress(meta *meta_v1.ObjectMeta) bool {
+// useOriginalSourceAddress returns true if the given object metadata indicates that the owner needs the Envoy listener to assume the identity of Cilium Ingress.
+// This can be an explicit label or the presence of an OwnerReference of Kind "Ingress" or "Gateway".
+func useOriginalSourceAddress(meta *meta_v1.ObjectMeta) bool {
 	for _, owner := range meta.OwnerReferences {
 		if owner.Kind == "Ingress" || owner.Kind == "Gateway" {
-			return true
+			return false
 		}
 	}
-	return false
+
+	if meta.GetLabels() != nil {
+		if v, ok := meta.GetLabels()["cilium.io/use-original-source-address"]; ok {
```

Comment:
the label name here is a little bit hidden, should we put it somewhere along with other constants :thinking:

Reply:
extracted constant into package `k8s`